index.js

```diff
@@ -21,436 +21,440 @@ const CB = {
+        data: results,
+        context: [
+          `FTS search - scoped to: inventory.hotel within fields ${cols.join(', ')}\n${JSON.stringify(qp)}`]
+      })
+    })
+  )
+
+  // Error handler. Must be defined after other routes/middleware
+  app.use((err, req, res, next) => {
+    const errText = err.toString()
+    if (errText.match(/LCB_ERR_KVENGINE_INVALID_PACKET/)) {
+      return res.status(500).send({
+        error: "Received LCB_ERR_KVENGINE_INVALID_PACKET error from Couchbase. Please check the SDK release notes and ensure you are using a compatible server version."
       })
     }
+    else {
+      return res.status(500).send({
+        error: `${err.toString()}: ${JSON.stringify(err)}`
+      })
+    }
+    next()
+  })
 
-    const addressCols = [
-      'address',
-      'state',
-      'city',
-      'country'
-    ]
-
-    const cols = [
-      'type',
-      'name',
-      'description',
-      ...addressCols
-    ]
-
-    const results = await Promise.all(
-      rows.map(async (row) => {
-        const doc = await hotels.get(row.id, {
-          project: cols
+  app.listen(8080, () => {
+    console.log(`Connecting to backend Couchbase server ${CB.host} with ${CB.username}/${CB.password}`)
+    console.log('Example app listening on port 8080!')
+  })
+}
+
+function authUser(req, res, next) {
+  bearerToken()(req, res, () => {
+    // Temporary Hack to extract the token from the request
+    req.token = req.headers.authorization.split(' ')[1]
+    jwt.verify(req.token, JWT_KEY, (err, decoded) => {
+      if (err) {
+        return res.status(400).send({
+          error: 'Invalid JWT token',
+          cause: err,
         })
-        var content = doc.content
-        content.address =
-          addressCols
-          .flatMap(field => content[field] || [])
-          .join(', ')
-        return content
-      })
-    )
+      }
 
-    return res.send({
-      data: results,
-      context: [
-        `FTS search - scoped to: inventory.hotel within fields ${cols.join(', ')}\n${JSON.stringify(qp)}`]
+      req.user = decoded
+      next()
     })
   })
-)
-
-// Error handler. Must be defined after other routes/middleware
-app.use((err, req, res, next) => {
-  const errText = err.toString()
-  if (errText.match(/LCB_ERR_KVENGINE_INVALID_PACKET/)) {
-    return res.status(500).send({
-      error: "Received LCB_ERR_KVENGINE_INVALID_PACKET error from Couchbase. Please check the SDK release notes and ensure you are using a compatible server version."
-    })
-  }
-  else {
-    return res.status(500).send({
-      error: `${err.toString()}: ${JSON.stringify(err)}`
-    })
+}
+
+function runAsync (callback) {
+  return function (req, res, next) {
+    callback(req, res, next)
+      .catch(next)
   }
-  next()
-})
+}
 
-app.listen(8080, () => {
-  console.log(`Connecting to backend Couchbase server ${CB.host} with ${CB.username}/${CB.password}`)
-  console.log('Example app listening on port 8080!')
-})
+main()
```

Comment:
Do we need to await main? I can't remember with Node.js async stuff 🧠 .

Reply:
`SyntaxError: await is only valid in async functions and the top level bodies of modules`